src/algorithm/neighbour/mod.rs

```diff
@@ -39,6 +39,10 @@ use serde::{Deserialize, Serialize};
 pub(crate) mod bbd_tree;
 /// tree data structure for fast nearest neighbor search
 pub mod cover_tree;
+/// structure for representing pairwise dissimilatiries (Müllner, 2011)
+pub(crate) mod dissimilarities;
+/// implementation for FastPair (Eppsterin, 2000) algorithm for Closest Pairs
+pub mod fastpair;
```

Comment:
We don't need to expose this class to the public, right? 
```suggestion
pub(crate) mod fastpair;
```